testing.rst

```diff
@@ -11,377 +11,513 @@ using both functional and unit tests.
-    it again. Another solution is to remove the project's ``symfony.lock`` file and
-    run ``composer install`` to force the execution of all Symfony Flex recipes.
+    :ref:`Symfony Flex <symfony-flex>` automatically creates
+    ``phpunit.xml.dist`` and ``tests/bootstrap.php``. If these files are
+    missing, you can try running the recipe again using
+    ``composer recipes:install phpunit/phpunit --force -v``.
 
-Each test - whether it's a unit test or a functional test - is a PHP class
-that should live in the ``tests/`` directory of your application. If you follow
-this rule, then you can run all of your application's tests with the same
-command as before.
+Types of Tests
+--------------
 
-PHPUnit is configured by the ``phpunit.xml.dist`` file in the root of your
-Symfony application.
+There are many types of automated tests and precise definitions often
+differ from project to project. In Symfony, the following definitions are
+used. If you have learned something different, that is not necessarily
+wrong, just different from what the Symfony documentation is using.
 
-.. tip::
+`Unit Tests`_
+    These tests ensure that *individual* units of source code (e.g. a single
+    class) behave as intended.
 
-    Use the ``--coverage-*`` command options to generate code coverage reports.
-    Read the PHPUnit manual to learn more about `code coverage analysis`_.
+`Integration Tests`_
+    These tests test a combination of classes and commonly interact with
+    Symfony's service container. These tests do not yet cover the fully
+    working application, those are called *Application tests*.
 
-.. index::
-   single: Tests; Unit tests
+`Application Tests`_
+    Application tests test the behavior of a complete application. They
+    make, sometimes fake, HTTP requests and test that the response is as
+    expected.
 
 Unit Tests
 ----------
 
-A `unit test`_ ensures that individual units of source code (e.g. a single class
-or some specific method in some class) meet their design and behave as intended.
-If you want to test an entire feature of your application (e.g. registering as a
-user or generating an invoice), see the section about :ref:`Functional Tests <functional-tests>`.
+A `unit test`_ ensures that individual units of source code (e.g. a single
+class or some specific method in some class) meet their design and behave
+as intended. Writing unit tests in a Symfony application no different from
+writing standard PHPUnit unit tests. You can learn about it in the PHPUnit
+documentation: `Writing Tests for PHPUnit`_.
 
-Writing Symfony unit tests is no different from writing standard PHPUnit
-unit tests. Suppose, for example, that you have an class called ``Calculator``
-in the ``src/Util/`` directory of the app::
+By convention, the ``tests/`` directory should replicate the directory
+of your application for unit tests. So, if you're testing a class in the
+``src/Util/`` directory, put the test in the ``tests/Util/`` directory.
```

Comment:
Could we find another name than `Util` for this directory? Such meaningless names are usually considered a bad practice: https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common